packages/language-server/src/server.ts

```diff
@@ -63,6 +64,22 @@ connection.onCompletionResolve(
   }
 );
 
+documents.onDidChangeContent(async changeEvent => {
+  // TODO: can we ensure the promise has always been initialized first (never be undefined)?
+  if (getSemanticModelPromise === undefined) {
+    return;
+  }
+  // TODO: how to debounce this? should we debounce this?
+  const ui5Model = await getSemanticModelPromise;
+  // TODO: should we check we are dealing with a *.[view|fragment].xml?
```

Comment:
It's done in the client... we could add it to the server too, not sure if we should

Reply:
maybe it is good as defensive coding if someone else implements a client and forgets to do so?

Reply:
I will update the comment and leave it in for now so we consider it later.